test/integration/builder/transaction.js

```diff
@@ -659,5 +659,34 @@ module.exports = function(knex) {
         return builder;
       });
     });
+
+    it('#3690 does not swallow exception when error during transaction occurs', async () => {
+      const killConnectionMap = {
+        mysql: async (connection) => knex.raw('KILL ?', [connection.threadId]),
+        mysql2: async (connection) => knex.raw('KILL ?', [connection.threadId]),
+        pg: async (connection) =>
+          knex.raw('SELECT pg_terminate_backend(?)', [connection.processID]),
+        redshift: async (connection) =>
+          knex.raw('SELECT pg_terminate_backend(?)', [connection.processID]),
+        mssql: async (connection, trx) => {
+          const id = (await trx.raw('select @@SPID as id'))[0].id;
+          return knex.raw(`KILL ${id}`);
+        },
+        oracledb: async (connection, trx) => {
+          const id = (await trx.raw(
+            `select sys_context('userenv','sessionid') id from dual;`
+          ))[0].id;
+          return knex.raw(`Alter System Kill Session '${id}'`);
+        },
+      };
+
+      const killConnection = killConnectionMap[knex.client.driverName];
+      if (!killConnection) return;
```

Comment:
If we use a standard function when declaring the `mocha` test, then we can do the following:

```suggestion
      if (!killConnection) {
        return this.skip();
      }
```

This will mark the test as "pending" rather than "succeeded" for the cases that we currently do not cover.

Reply:
done